engine.go

```diff
@@ -365,36 +365,36 @@ func bindingsToExprs(bindings map[string]*querypb.BindVariable) (map[string]sql.
 
 // QueryWithBindings executes the query given with the bindings provided.
 // If parsed is non-nil, it will be used instead of parsing the query from text.
-func (e *Engine) QueryWithBindings(ctx *sql.Context, query string, parsed sqlparser.Statement, bindings map[string]*querypb.BindVariable, qFlags *sql.QueryFlags) (sql.Schema, sql.RowIter, error) {
+func (e *Engine) QueryWithBindings(ctx *sql.Context, query string, parsed sqlparser.Statement, bindings map[string]*querypb.BindVariable, qFlags *sql.QueryFlags) (sql.Schema, sql.RowIter, *sql.QueryFlags, error) {
 	sql.IncrementStatusVariable(ctx, "Questions", 1)
 
 	query = sql.RemoveSpaceAndDelimiter(query, ';')
 
 	parsed, binder, err := e.preparedStatement(ctx, query, parsed, bindings)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	// Give the integrator a chance to reject the session before proceeding
 	// TODO: this check doesn't belong here
 	err = ctx.Session.ValidateSession(ctx)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	err = e.beginTransaction(ctx)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	bound, qFlags, err := e.bindQuery(ctx, query, parsed, bindings, binder, nil)
```

Comment:
Should the value of `qFlags` that the caller passes in be passed to `bindQuery`? Otherwise it seems like the passed in `Flags` value is never used anywhere?

Reply:
Yes will fix. At the meta level there was two ways to go about this, which was being very narrow with where we add the argument, or being very permissive with where we add the argument. I leaned towards permissive, because most of the interfaces already seem permissive for binding early or late. To test this intentionally rather than implicitly with the server interface, we would need a way to enable or disable the use of the new flag context.